### PR TITLE
Changes to late declaration related use cases

### PR DESCRIPTION
**Pull request by notadragon**

### CODE REVIEW

File: contracts_use_cases.html [hunk 1]
```diff
@@ -268,6 +268,7 @@ <h2>Use cases &mdash; table</h2>
   <td>Make reference to the before and after values of mutable class members (ie. <i>new_size = old_size+1</i> after push_back) in post-conditions</td>
   <td></td></tr>
 
+
```
Comment (sempuki) [inline contracts_use_cases.html]: Several whitespace cleanups.

File: contracts_use_cases.html [hunk 2]
```diff
@@ -366,8 +367,15 @@ <h2>Use cases &mdash; table</h2>
 
 <tr><td></td>
   <td>API Developer</td>
-  <td>Keep my user interfaces clean and narrow</td>
-  <td>Isolate users reading my interface from the specific implementations of my contract, because they are expressed using implementation details and my English language contract is more readable</td>
+  <td>Keep my public interfaces clean and concise</td>
+  <td>Place member function contract conditions outside the class definition (e.g., because my English-language contracts are more
```
Comment (sempuki) [inline contracts_use_cases.html]: Did you mean `outside class *declaration*`?

File: contracts_use_cases.html [hunk 3]
```diff
@@ -366,8 +367,15 @@ <h2>Use cases &mdash; table</h2>
...
+    complete and more readable, or because the conditions might reference implementation details not relevant to readers).</td>
+  <td></td></tr>
+
+<tr><td></td>
+  <td>API Developer</td>
+  <td>Keep my public interfaces clean and concise</td>
+  <td>Place function contract conditions after the first declaration (e.g., because my English-language contracts are more complete and more readable, and my contract conditions are implementation details).</td>
```
Comment (sempuki) [inline contracts_use_cases.html]: This doesn't say why you want it in the declaration or the first declaration.

File: contracts_use_cases.html (hunk 2 above)
Comment (sempuki) [reply contracts_use_cases.html]: We haven't discussed `classes` in this part of the use cases, and if we did they'd be in `C++ developer`. What is the essence of the distinction you're trying to make?

File: contracts_use_cases.html [hunk 4]
```diff
@@ -660,14 +668,15 @@ <h2>Use cases &mdash; table</h2>
 
 <tr><td></td>
   <td>C++ Library Developer</td>
-  <td>Reduce need for client recompilation</td>
-  <td>Insulate contract conditions with the function body</td>
+  <td>Control the tradeoff between need for client recompilation and contract condition visibility</td>
+  <td>Insulate contract conditions with the function definition</td>
```
Comment (sempuki) [inline contracts_use_cases.html]: In this case the `I want to` does not lead to `In order to`, in the sense that it doesn't allow increasing "contract condition visibility".

File: contracts_use_cases.html [hunk 5]
```diff
@@ -660,14 +668,15 @@ <h2>Use cases &mdash; table</h2>
...
   <td></td></tr>
 
 <tr><td></td>
   <td>C++ Library Developer</td>
-  <td>Reduce need for client recompilation</td>
-  <td>Support traditional separation of definition and declaration</td>
+  <td>Control the tradeoff between need for client recompilation and contract condition visibility</td>
+  <td>Insulate the function body but not the contract conditions (e.g., so static analysis tools can view the contract conditions without
+    needing to process the function body) but still outside the class definition to avoid cluttering my public interface.
```
Comment (sempuki) [inline contracts_use_cases.html]: Why is this `I want to` C++ class specific?

**Comment by sempuki** [comment]
Aside from some clear improvements, like replacing `and` with `or`, and `body` with `definition`, I fail to understand how the replacement wording improves things, or what distinctions it's trying to make. I want to warn against adding too many words as it may be defeating your own purpose: people are going to either fail to read long sentences, or misunderstand what those sentences are driving at, and deprioritize those cases.

If you'd like me to try to respecify them I would be willing. If you'd like them in verbatim I'm ok with that as well, since it doesn't harm my priority use cases.

**Review by sempuki — CHANGES_REQUESTED** [review]

File: contracts_use_cases.html (hunk 5 above)
Comment (notadragon) [reply contracts_use_cases.html]: Because noise.  I'm going to trim this down.   Part of this is to motivate a need to redeclare member functions so that contract conditions on member functions can be put outside of the class definition but still "at the bottom of the header" where they won't clutter the human-readable part of the header.

**Comment by notadragon** [comment]
I pushed in a new change with what we think is more concise wording.  Let me know if that's clearer and i'll merge.
